configure.ac

```diff
@@ -7,14 +7,52 @@ AC_CONFIG_HEADERS([config.h])
 AC_PROG_CXX
 AM_PROG_AR
 
+AC_CANONICAL_HOST
+
+build_linux=no
+build_windows=no
+build_mac=no
+
+# Detect the target system
+case "${host_os}" in
+    linux*)
+        build_linux=yes
+        ;;
+    cygwin*|mingw*)
+        build_windows=yes
+        ;;
+    darwin*)
+        AC_MSG_ERROR(["OS $host_os is not supported"])
+        ;;
+    *)
+        AC_MSG_ERROR(["OS $host_os is not supported"])
+        ;;
+esac
+
+# Pass the conditionals to automake
+AM_CONDITIONAL([LINUX], [test "$build_linux" = "yes"])
+AM_CONDITIONAL([WINDOWS], [test "$build_windows" = "yes"])
+AM_CONDITIONAL([OSX], [test "$build_mac" = "yes"])
+
 AC_LANG([C++])
 LT_INIT
 
+# Windows mu have no undefined symbol in dll, we need -no-undefined 
+# flag to work, so lets trigger clean dll build
+
+AM_COND_IF( [WINDOWS], [
+LT_INIT([win32-dll])
+])
```

Comment:
In the comment, you mention the `-no-undefined` flag - how is that related to the actual command later on?  Does `win32-dll` set this?  Also, as a small nit, please remove the blank line between the comment and the code below.

Reply:
I am not sure, I remember I only got correct results with combination of win32-dll being set there and -no-undefined later on. I will try removing those now and see, if that is really necessary. If so, so I will supply error log with explanations.

Reply:
Ok.  If you need indeed both, then let me know.  I don't think you need to post detailed error logs - as long as you are confident that both are needed, I'll trust you on that. :)